### ISSUE TITLE
ConfigValidator validation failures are overridden by subsequent successes

### ISSUE THREAD

**Issue by erademacher** [hashicorp/terraform-plugin-framework#635]
### Module version

```
github.com/hashicorp/terraform-plugin-framework v1.0.1
```

### Relevant provider source code

```go
func (r *clusterResource) ConfigValidators(_ context.Context) []resource.ConfigValidator {
	return []resource.ConfigValidator{
		resourcevalidator.Conflicting(
			path.MatchRoot("dedicated"),
			path.MatchRoot("serverless"),
		),
		resourcevalidator.Conflicting(
			path.MatchRoot("dedicated").AtName("num_virtual_cpus"),
			path.MatchRoot("dedicated").AtName("machine_type"),
		),
	}
}
```

### Terraform Configuration Files

```hcl
  dedicated = {
    storage_gib  = var.storage_gib
    num_virtual_cpus = var.vcpus
  }
  serverless = {
    spend_limit: 0,
  }
```

### Expected Behavior
Validation should fail and return an error since the first ConfigValidator failed.

### Actual Behavior
Validation succeeds because the last ConfigValidator succeeded. Output from earlier validators is lost.

### Steps to Reproduce
1. `terraform validate`

---

In `server_validateresourceconfig.go`, `ValidateResourceConfig` loops through the resource's `ConfigValidator`s. `ValidateResourceConfig`'s output diagnostics are _set_ to each individual validator's diagnostics, not appended. They should probably be appended instead.

**Comment by bflad** [comment]
Hi @erademacher 👋 Thank you for reporting this and sorry you ran into trouble here.

Can you try upgrading terraform-plugin-framework to the latest? [Version 1.1.0](https://github.com/hashicorp/terraform-plugin-framework/releases/tag/v1.1.0) should contain this bug fix:

> resource: Prevented ConfigValidators from unexpectedly modifying or removing prior validator diagnostics

Thanks.

**Comment by erademacher** [comment]
Sorry! I thought I was on the latest version. This issue is indeed fixed in 1.1.1, but it revealed another one. Using the same provider code, if I remove `serverless` from the config but add `machine_type` to `dedicated`, `terraform validate` succeeds but `plan` and `apply` fail with the correct validation error. I'll open a new issue.
